sdk/core/azure-core/src/main/java/com/azure/core/handler/JUL2SLF4JHandler.java

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.handler;
+
+import java.text.MessageFormat;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.ResourceBundle;
+import java.util.MissingResourceException;
+
+/**
+ * This is bridge/route all JUL log records to the SLF4J API.
+ *
+ * Essentially, the idea is to install on the root logger an instance of JUL2SLF4JHandler
+ * as the sole JUL handler in the system.
+ * Subsequently, the JUL2SLF4JHandler instance will redirect all JUL log records are redirected to
+ * the SLF4J API based on the following mapping of levels:
+ * FINEST -&gt; TRACE
+ * FINER -&gt; DEBUG
+ * FINE -&gt; DEBUG
+ * INFO -&gt; INFO
+ * WARNING -&gt; WARN
+ * SEVERE -&gt; ERROR
+ *
+ * <p>Programmatic installation:</p>
+ * Optionally remove existing handlers attached to j.u.l root logger
+ * JUL2SLF4JHandler.removeHandlersForRootLogger()
+ *
+ * add JUL2SLF4JHandler to j.u.l's root logger, should be done once during
+ * the initialization phase of your application
+ * JUL2SLF4JHandler.install();
+ *
+ * <p>Installation via logging.properties configuration file:</p>
+ * register JUL2SLF4JHandler as handler for the j.u.l. root logger
+ * handlers = com.azure.core.handler.JUL2SLF4JHandler
+ *
+ * Once JUL2SLF4JHandler is installed, logging by j.u.l. loggers will be directed to SLF4J. Example:
+ * import  java.util.logging.Logger;
+ * ...
+ * usual pattern: get a Logger and then log a message
+ * Logger julLogger = Logger.getLogger("org.wombat");
+ * julLogger.fine("hello world"); // this will get redirected to SLF4J
+ *
+ * Please note that translating a java.util.logging event into SLF4J incurs the cost of constructing LogRecord
+ * instance regardless of whether the SLF4J logger is disabled for the given level.
+ * Consequently, j.u.l. to SLF4J translation can seriously increase the cost of disabled logging statements
+ * (60 fold or 6000% increase) and measurably impact the performance of enabled log statements (20% overall increase).
+ * Please note that as of logback-version 0.9.25, it is possible to completely eliminate the 60 fold translation
+ * overhead for disabled log statements with the help of <a href="http://logback.qos.ch/manual/configuration.html#LevelChangePropagator">LevelChangePropagator</a>.
+ *
+ */
+public class JUL2SLF4JHandler extends java.util.logging.Handler {
```

Comment:
@JonathanGiles I wanted your opinion on this since it is in the public API for azure-core.

The reason for this is that an external dependency we use (ie. proton-j) leverages `java.util.logger` while we use slf4j. So when we are looking at support cases, we ask them to enable AZURE_LOG_LEVEL, which gives insights into our client libraries but not for proton-j. So it would be nice to ask them to enable a single configuration so we do not have to get them to add two logging implementations.

Reply:
I have to say that on initial inspection this PR makes me uneasy. It appears to me that the problem we are trying to solve is one where the logging output from a library is not aligned with our logging infrastructure, and so we are asking users to install this class to hijack the logs and route them out through our output. I wonder why this is better than asking the user to configure the proton-j logging as necessary?

Reply:
That's fair. Thanks for that perspective on this. Thinking more about this, it may not be the best solution. 🤔 @YijunXieMS Thoughts?